File: mobile/src/main/java/treehou/se/habit/ui/util/ViewHelper.java
package treehou.se.habit.ui.util;

import android.graphics.PorterDuff;
import android.widget.RemoteViews;

import java.lang.reflect.InvocationTargetException;
import java.lang.reflect.Method;

/**
 * Contains code helping to create views.
 */
public class ViewHelper {

    private ViewHelper() {}

    public static void colorRemoteDrawable(RemoteViews cellView, int resource, int color){
    try {
        Class c = Class.forName("android.widget.RemoteViews");
        Method m = c.getMethod("setDrawableParameters", int.class, boolean.class, int.class, int.class, PorterDuff.Mode.class, int.class);
        m.invoke(cellView, resource, true, -1, color, PorterDuff.Mode.MULTIPLY, -1);
        } catch (ClassNotFoundException e) {
            e.printStackTrace();
        } catch (InvocationTargetException e) {
            e.printStackTrace();
        } catch (NoSuchMethodException e) {
            e.printStackTrace();
        } catch (IllegalAccessException e) {
            e.printStackTrace();
        }
    }
}

File: mobile/src/main/java/treehou/se/habit/ui/util/ViewHelper.kt
package treehou.se.habit.ui.util

import android.graphics.PorterDuff
import android.widget.RemoteViews

import java.lang.reflect.InvocationTargetException
import java.lang.reflect.Method

/**
 * Contains code helping to create views.
 */
object ViewHelper {

    fun colorRemoteDrawable(cellView: RemoteViews, resource: Int, color: Int) {
        try {
            val c = Class.forName("android.widget.RemoteViews")
            val m = c.getMethod("setDrawableParameters", Int::class.javaPrimitiveType, Boolean::class.javaPrimitiveType, Int::class.javaPrimitiveType, Int::class.javaPrimitiveType, PorterDuff.Mode::class.java, Int::class.javaPrimitiveType)
            m.invoke(cellView, resource, true, -1, color, PorterDuff.Mode.MULTIPLY, -1)
        } catch (e: ClassNotFoundException) {
            e.printStackTrace()
        } catch (e: InvocationTargetException) {
            e.printStackTrace()
        } catch (e: NoSuchMethodException) {
            e.printStackTrace()
        } catch (e: IllegalAccessException) {
            e.printStackTrace()
        }

    }
}
